tools/loadtesting/terraform/ecs.tf

```diff
@@ -0,0 +1,388 @@
+          },
+          {
+            name  = "FLEET_REDIS_ADDRESS"
+            value = "${aws_elasticache_replication_group.default.primary_endpoint_address}:6379"
+          },
+          {
+            name  = "FLEET_REDIS_CLUSTER_FOLLOW_REDIRECTIONS"
+            value = "true"
+          },
+          {
+            name  = "FLEET_FIREHOSE_STATUS_STREAM"
+            value = aws_kinesis_firehose_delivery_stream.osquery_status.name
+          },
+          {
+            name  = "FLEET_FIREHOSE_RESULT_STREAM"
+            value = aws_kinesis_firehose_delivery_stream.osquery_results.name
+          },
+          {
+            name  = "FLEET_FIREHOSE_REGION"
+            value = data.aws_region.current.name
+          },
+          {
+            name  = "FLEET_OSQUERY_STATUS_LOG_PLUGIN"
+            value = "firehose"
+          },
+          {
+            name  = "FLEET_OSQUERY_RESULT_LOG_PLUGIN"
+            value = "firehose"
+          },
+          {
+            name  = "FLEET_SERVER_TLS"
+            value = "false"
+          },
+          {
+            name  = "FLEET_REDIS_MAX_IDLE_CONNS"
+            value = "100"
+          },
+          {
+            name  = "FLEET_REDIS_MAX_OPEN_CONNS"
+            value = "100"
+          },
+          {
+            name  = "FLEET_OSQUERY_ASYNC_HOST_REDIS_SCAN_KEYS_COUNT"
+            value = "10000"
+          }
+        ], local.additional_env_vars)
+      }
+  ])
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+
+resource "aws_ecs_task_definition" "migration" {
+  family                   = "fleet-migrate"
+  network_mode             = "awsvpc"
+  requires_compatibilities = ["FARGATE"]
+  execution_role_arn       = aws_iam_role.main.arn
+  task_role_arn            = aws_iam_role.main.arn
+  cpu                      = 1024
+  memory                   = 2048
+  container_definitions = jsonencode(
+    [
+      {
+        name        = "fleet-prepare-db"
+        image       = docker_registry_image.fleet.name
+        cpu         = 1024
+        memory      = 2048
+        mountPoints = []
+        volumesFrom = []
+        essential   = true
+        portMappings = [
+          {
+            # This port is the same that the contained application also uses
+            containerPort = 8080
+            protocol      = "tcp"
+          }
+        ]
+        networkMode = "awsvpc"
+        logConfiguration = {
+          logDriver = "awslogs"
+          options = {
+            awslogs-group         = aws_cloudwatch_log_group.backend.name
+            awslogs-region        = data.aws_region.current.name
+            awslogs-stream-prefix = "fleet"
+          }
+        },
+        command = ["fleet", "prepare", "--no-prompt=true", "db"]
+        secrets = [
+          {
+            name      = "FLEET_MYSQL_PASSWORD"
+            valueFrom = aws_secretsmanager_secret.database_password_secret.arn
+          }
+        ]
+        environment = [
```

Comment:
Should we also set `CLOUDWATCH_NAMESPACE` to something specific to migrations?

Reply:
Yes we should, meant to do that.

Reply:
fixed